pkg/transformer/eth_getTransactionByHash.go

```diff
@@ -19,109 +20,209 @@ func (p *ProxyETHGetTransactionByHash) Method() string {
+	if err != nil {
 		return nil, err
 	}
+	return ethTx, nil
+}
 
-	ethVal, err := QtumAmountToEthValue(tx.Amount)
+// TODO: think of returning flag if it's a rewrad transaction for miner
+func getTransactionByHash(p *qtum.Qtum, hash string) (*eth.GetTransactionByHashResponse, error) {
+	qtumTx, err := p.GetTransaction(hash)
 	if err != nil {
+		// TODO: implement typed error at Qtum pkg
+		// * Filter "transaction not found" error
+		//
+		// TODO: researching
+		// ?! Is the case only for reward transactions
+		cause := errors.Cause(err)
+		if !strings.Contains(cause.Error(), "[code: -5]") {
+			return nil, err
+		}
 
-		return nil, err
+		ethTx, err := getRewardTransactionByHash(p, hash)
+		if err != nil {
+			return nil, errors.WithMessage(err, "couldn't get reward transaction by hash")
+		}
+		return ethTx, nil
+	}
+	qtumDecodedRawTx, err := p.DecodeRawTransaction(qtumTx.Hex)
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't get raw transaction")
+	}
+
+	ethTx := &eth.GetTransactionByHashResponse{
+		Hash: utils.AddHexPrefix(qtumDecodedRawTx.ID),
+		// TODO: researching
+		// ? May be don't need this
+		// ! Probably, needs huge quantity requests to calc by hands
+		Nonce: "0x0",
+
+		// TODO: researching
+		// ? Do we need those values
+		V: "",
+		R: "",
+		S: "",
 	}
 
-	decodedRawTx, err := p.Qtum.DecodeRawTransaction(tx.Hex)
+	if !qtumTx.IsPending() { // otherwise, the following values must be nulls
+		blockNumber, err := getBlockNumberByHash(p, qtumTx.BlockHash)
+		if err != nil {
+			return nil, errors.WithMessage(err, "couldn't get block number by hash")
+		}
+		ethTx.BlockNumber = hexutil.EncodeUint64(blockNumber)
+		ethTx.BlockHash = utils.AddHexPrefix(qtumTx.BlockHash)
+		ethTx.TransactionIndex = hexutil.EncodeUint64(uint64(qtumTx.BlockIndex))
+	}
+
+	ethAmount, err := formatQtumAmount(qtumDecodedRawTx.CalcAmount())
 	if err != nil {
-		return nil, errors.Wrap(err, "Qtum#DecodeRawTransaction")
+		return nil, errors.WithMessage(err, "couldn't format amount")
 	}
+	ethTx.Value = ethAmount
 
-	ethTx := eth.GetTransactionByHashResponse{
-		Hash:      utils.AddHexPrefix(tx.Txid),
-		BlockHash: utils.AddHexPrefix(tx.Blockhash),
-		Nonce:     "0x01",
-		Value:     ethVal,
+	qtumTxContractInfo, isContractTx, err := qtumDecodedRawTx.ExtractContractInfo()
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't extract contract info")
+	}
+	if isContractTx {
+		ethTx.Input = utils.AddHexPrefix(qtum.ZeroUserInput)
+		if qtumTxContractInfo.UserInput == "" {
+			ethTx.Input = utils.AddHexPrefix(qtumTxContractInfo.UserInput)
+		}
+		ethTx.From = utils.AddHexPrefix(qtumTxContractInfo.From)
+		ethTx.To = utils.AddHexPrefix(qtumTxContractInfo.To)
+		ethTx.Gas = utils.AddHexPrefix(qtumTxContractInfo.GasUsed)
+		// TODO: discuss, consider values
+		// - gas limit is needed for `eth_getBlockByHash` request (return as a func result?)
+		// ? ethTx.GasPrice
+		ethTx.GasPrice = "0x0" // temporary solution
```

Comment:
Gas Price is generally 0x40. But I *believe* this should be returnable?

Reply:
Issue is resolved, see commit by hash: cd94946d7db645503a58d64a794397dfd68b772f